### PR TITLE
Determine JAVA_HOME on Debian/Ubuntu

### PR DESCRIPTION
**Pull request by terokorp**

### CODE REVIEW

File: src/release/bin/startup.sh [hunk 1]
```diff
@@ -5,6 +5,11 @@
 # $Id$
 # -----------------------------------------------------------------------------
 
+# Determine JAVA_HOME on Debian/Ubuntu
+if [ -z "$JAVA_HOME" ]; then
+  JAVA_HOME=$(readlink -fe /usr/bin/java | sed "s:bin/java::")
```
Comment (bencaradocdavies) [inline src/release/bin/startup.sh]: This is elegant because, if the link is missing or broken, ```JAVA_HOME``` will stay empty. I like this fix because I am always annoyed by having to manually set ```JAVA_HOME``` when starting the bin.zip.

My only remark is that, in ```readlink -fe```, the "f" is redundant.

**Comment by bencaradocdavies** [comment]
Another problem is that ```readlink``` on MacOS does supports neither ```-f``` nor ```-e```. Also, ```sed``` on MacOS has some eccentricities.

Another solution (non-MacOS) is ```$(dirname $(dirname $(readlink -f $(which java))))```, with similar MacOS version here:
See http://stackoverflow.com/questions/1117398/java-home-directory

**Comment by bencaradocdavies** [comment]
TL;DR: great idea, not quite ready as would break MacOS.

**Comment by bencaradocdavies** [comment]
A better question is, why do we even want to determine ```JAVA_HOME```? It is only used to locate ```java```. Once ```java``` is running, it can knows its own ```System.getProperty("java.home")``` and AFAIK requires no environment variables. We could just modify ```startup.sh``` to use ```_RUNJAVA=java``` if ```$(which java)``` is nonzero and only consider ```JAVA_HOME``` otherwise.

**Comment by bencaradocdavies** [comment]
I have an alternative solution in https://github.com/geoserver/geoserver/pull/1553 which avoids setting ```JAVA_HOME``` and falls back to the path. It also fixes ```shutdown.sh```.

**Comment by terokorp** [comment]
Yea, #1553 seems to be better. I will close this one.

**Comment by bencaradocdavies** [comment]
@terokorp, #1553 was just merged by @jodygarnett. Please try a subsequent nightly build (or 2.9-beta2 coming very soon) to see if it works for you. Thank you for your pull request, even though it was not merged. This was a necessary fix and we needed your prompting.  :-)
